src/systems/components-queue.js

```diff
@@ -0,0 +1,51 @@
+AFRAME.registerSystem("components-queue", {
+  init() {
+    // Registry a map from queue name to another map, which is a map from component object to its assigned index.
+    // Indicies range from 0 to N - 1 where N is the number of registered components in the queue.
+    // The shouldTick method returns true for the component whose index is the mod of the current frame index.
+    this.registry = new Map();
+    this.componentToQueue = new Map();
+    this.frameIndex = 0;
+  },
+
+  register(component, queue = component.name) {
```

Comment:
This is nice, eventually I could see us adding different strategies for each queue or even each item such that we could do things like specify a maximum frame delay and such.